src/witan/send/adroddiad/clerk/charting_v2.clj

```diff
@@ -0,0 +1,147 @@
+(ns witan.send.adroddiad.clerk.charting-v2
+  (:require
+   [clojure2d.color :as color]
+   [nextjournal.clerk :as clerk]
+   [tablecloth.api :as tc]))
+
+
+(defn color-and-shape-lookup [domain]
+  (tc/dataset
+   {:domain-value domain
+    :color (cycle 
+            ;; I’d like to eventually do something based on these colours in v3
+            ;; ["#fa814c" "#256cc6" "#fbe44c" "#50b938" "#59c4b8" "#29733c"]
+            (into []
+                  (map color/format-hex)
+                  (color/palette :tableau-20)))
+    :shape (cycle ["circle", "square", "cross", "diamond", "triangle-up", "triangle-down", "triangle-right", "triangle-left"])}))
+
+
+
+(defn color-map [plot-data color-field color-lookup]
+  (let [group-keys (into (sorted-set) (get plot-data color-field))
+        filtered-colors (tc/select-rows color-lookup #(group-keys (get % :domain-value)))]
+    {:field color-field
+     :scale {:range (into [] (:color filtered-colors))
+             :domain (into [] (:domain-value filtered-colors))}}))
+
+(defn shape-map [plot-data shape-field shape-lookup]
+  (let [group-keys (into (sorted-set) (get plot-data shape-field))
+        filtered-shapes (tc/select-rows shape-lookup #(group-keys (get % :domain-value)))]
+    {:field shape-field
+     :scale {:range (into [] (:shape filtered-shapes))
+             :domain (into [] (:domain-value filtered-shapes))}}))
+
+(defn line-plot
+  [{:keys [data
+           chart-title
+           chart-height chart-width
+           clerk-width
+           x x-title x-format
+           y y-title y-format y-zero
+           group group-title
+           colors-and-shapes]
```

Comment:
Is it worth having some more of these having defaults? I was just thinking as there was so many to consider

Reply:
hmm... possibly. Good idea for some additions